### PR TITLE
[QHC-32] QDevil QDAC-II

### PR DESCRIPTION
**Pull request by fedonman**

### CODE REVIEW

**Comment by AlbertMitjans** [comment]
Hello. You may have forgotten to update the changelog!
Please edit [`changelog-dev.md`](/qilimanjaro-tech/qililab/tree/main/doc/releases/changelog-dev.md) with:
* A one-to-two sentence description of the change. You may include a small working example for new features.
* A link back to this PR.

**Review by jjmartinezQT — CHANGES_REQUESTED** [review]
A minor change in docstrings. Victor will finish this review by testing it on hardware. Double check when testing this one on hardware the issue when disconnecting platform and not turning off instruments, so we can better characterise the error hardware was having with the Yokowaga.

File: tests/instrument_controllers/qdevil/test_qdevil_qdac2_controller.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+from qililab.platform import Platform
+from qililab.settings import Settings
+from tests.data import SauronQDevil  # pylint: disable=import-error
+from tests.test_utils import build_platform  # pylint: disable=import-error
+
+
+@pytest.fixture(name="platform")
+def fixture_platform() -> Platform:
+    """Return Platform object."""
+    return build_platform(runcard=SauronQDevil.runcard)
```
Comment (visagim) [inline tests/instrument_controllers/qdevil/test_qdevil_qdac2_controller.py:17]: perhaps it'd be better to have the runcard being mocked elsewhere (check https://qilimanjaro.slack.com/archives/C05D7KW4B6G/p1702912592701119)

File: tests/instrument_controllers/qdevil/test_qdevil_qdac2_controller.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    @patch("qililab.instrument_controllers.qdevil.qdevil_qdac2_controller.QDevilQDac2Device")
+    @pytest.mark.parametrize("controller_alias", ["qdac_controller_ip", "qdac_controller_usb"])
+    def test_initialize_device(self, device_mock: MagicMock, platform: Platform, controller_alias: str):
+        """Test QDAC-II controller initializes device correctly."""
+        controller_instance = platform.instrument_controllers.get_instrument_controller(alias=controller_alias)
+
+        controller_instance._initialize_device()
+
+        assert isinstance(controller_instance.device, MagicMock)
```
Comment (visagim) [inline tests/instrument_controllers/qdevil/test_qdevil_qdac2_controller.py:52]: check that the arguments given to `QDevilQDac2Device()` in:
```
        if self.settings.connection.name == ConnectionName.TCP_IP:
            self.device = QDevilQDac2Device(f"{self.name.value}", f"TCPIP::{self.address}::5025::SOCKET")
        else:
            self.device = QDevilQDac2Device(f"{self.name.value}", f"ASRL/dev/{self.address}::INSTR")
```
are as intended (perhaps we can leave the check that instance created is `MagicMock`, but I don't think that alone is enough)

**Review by visagim — COMMENTED** [review]
Cool! Just some small comments

File: tests/instrument_controllers/qdevil/test_qdevil_qdac2_controller.py (hunk 2 above)
Comment (fedonman) [reply tests/instrument_controllers/qdevil/test_qdevil_qdac2_controller.py:52]: I don't know how to do that tbh. I haven't managed to assert a constructor call so far.

Comment (fedonman) [reply tests/instrument_controllers/qdevil/test_qdevil_qdac2_controller.py]: I added checks for constructor parameters

**Review by visagim — APPROVED** [review]
Cool! It's good to go

**Review by jjmartinezQT — APPROVED** [review]
Approved.

**Pull request merged**